src/Mvc/Mvc.RazorPages/src/Infrastructure/CompiledPageActionDescriptorProvider.cs

```diff
@@ -82,4 +85,16 @@ public void OnProvidersExecuting(ActionDescriptorProviderContext context)
     public void OnProvidersExecuted(ActionDescriptorProviderContext context)
     {
     }
+
+    private static void EnsureApplicationPartsPopulated(ApplicationPartManager applicationPartManager)
+    {
+        if (applicationPartManager.EntryAssembly == null && !applicationPartManager.ApplicationParts!.Any(part => part is IRazorCompiledItemProvider))
```

Comment:
@pranavkm This is my biggest concern about this PR. I am trying to find a way to identify misconfigured environment, and this is one of the scenarios detected in the open issue, so, since we cannot have 100% certain since we could have no `IRazorCompiledItemProvider`,  I conditioned that with the `entryAssembly` as well because in this case probably we are missing something.